lib/gcloud/resource_manager/manager.rb

```diff
@@ -0,0 +1,318 @@
+#--
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "gcloud/resource_manager/credentials"
+require "gcloud/resource_manager/connection"
+require "gcloud/resource_manager/errors"
+require "gcloud/resource_manager/project"
+
+module Gcloud
+  module ResourceManager
+    ##
+    # = Manager
+    #
+    #   require "gcloud"
+    #
+    #   gcloud = Gcloud.new
+    #   resource_manager = gcloud.resource_manager
+    #   resource_manager.projects.each do |project|
+    #     puts projects.project_id
+    #   end
+    #
+    # See Gcloud#resource_manager
+    class Manager
+      ##
+      # The Connection object.
+      attr_accessor :connection #:nodoc:
+
+      ##
+      # Creates a new Connection instance.
+      #
+      # See Gcloud.resource_manager
+      def initialize credentials #:nodoc:
+        @connection = Connection.new credentials
+      end
+
+      ##
+      # Retrieves the projects that are visible to the user and satisfy the
+      # specified filter. This method returns projects in an unspecified order.
+      # New projects do not necessarily appear at the end of the list.
+      #
+      # === Parameters
+      #
+      # +filter+::
+      #   An expression for filtering the results of the request. Filter rules
+      #   are case insensitive. (+String+)
+      #
+      #   The fields eligible for filtering are:
+      #   * +name+
+      #   * +id+
+      #   * +labels.key+ - where +key+ is the name of a label
+      #
+      #   Some examples of using labels as filters:
+      #   * +name:*+ - The project has a name.
+      #   * +name:Howl+ - The project's name is Howl or howl.
+      #   * +name:HOWL+ - Equivalent to above.
+      #   * +NAME:howl+ - Equivalent to above.
+      #   * +labels.color:*+ - The project has the label color.
+      #   * +labels.color:red+ - The project's label color has the value red.
+      #   * +labels.color:red label.size:big+ - The project's label color has
```

Comment:
That documentation comes from the [Google Cloud Resource Manager](https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects/list#body.QUERY_PARAMETERS.filter) docs. So if it is wrong, then it looks to be wrong on the official docs.

I am stuck on permissions issues trying to make updates to projects, but as soon as I resolve that I'll verify whether both `labels` and `label` are supported on the filter.